src/Symfony/Component/Validator/Constraints/Base64.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+final class Base64 extends Constraint
+{
+    public const INVALID_STRING_ERROR = '5699d0ba-3cea-4676-9aab-d3303ebd6934';
+
+    protected const ERROR_NAMES = [
+        self::INVALID_STRING_ERROR => 'INVALID_STRING_ERROR',
+    ];
+
+    public bool $allowDataUri = false;
+    public string $message = 'The given string is not a valid Base64 encoded string.';
+
+    public function __construct(bool $allowDataUri = false, string $message = null, array $groups = null, mixed $payload = null, array $options = null)
+    {
+        parent::__construct($options, $groups, $payload);
+
+        $this->message = $message ?? $this->message;
+        $this->allowDataUri = $allowDataUri;
```

Comment:
I'm not sure I understand how this is supposed to be useful.
Do you really have a use case where data URLs are accepted as input?
The main issue I see is the fact that the URL prefix is optional when enabling this: either you accept without the prefix, or you accept with the prefix, but both? really?

Reply:
About the use case, yes: we have use cases where a user uploads, let's say, a PNG. The input data looks like `data:image/png;base64,iVBORw0KGgo...`. However, we also accept, in other DTOs, "simple" base64 without data uri. I think giving this option helps the constraint being flexible for different cases devs will encounter.

That said, I agree that accepting both is useless. It is either one case or another. I'll update the code to have an option like `requiresDataUri` which makes more sense.